src/lib/Builder.hs

```diff
@@ -261,13 +262,24 @@ class (EnvReader m, MonadFail1 m)
   -- `emitBinding` is expressible in terms of `emitEnv` but it can be
   -- implemented more efficiently by avoiding a double substitution
   -- XXX: emitBinding/emitEnv don't extend the synthesis candidates
+  -- TODO: do we actually need `emitBinding`? Top emissions probably aren't hot.
```

Comment:
Just that `emitBinding` is just an optimized special case of `emitEnv`. But the optimization is probably unnecessary because we only use it for top-level emissions, which are much less common that local emissions.